File: daily/Linked List/2487-Remove-nodes-from-linked-list.py
class Solution:
    def removeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None
        stack = deque()
        curr = head
        while curr:
            while stack and stack[-1].val < curr.val:
                stack.pop()
            stack.append(curr)
            curr = curr.next
        
        new_head = stack[0]
        curr_node = new_head
        for i in range(1, len(stack)):
            curr_node.next = stack[i]
            curr_node = curr_node.next
        curr_node.next = None
        return new_head

File: daily/Linked List/2487-Remove-nodes-from-linked-list.go
package daily

func removeNodes(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}
	stack := []*ListNode{}
	curr := head
	for curr != nil {
		for len(stack) > 0 && stack[len(stack)-1].Val < curr.Val {
			stack = stack[:len(stack)-1]
		}
		stack = append(stack, curr)
		curr = curr.Next
	}

	newHead := stack[0]
	currNode := newHead
	for i := 1; i < len(stack); i++ {
		currNode.Next = stack[i]
		currNode = currNode.Next
	}
	currNode.Next = nil
	return newHead
}
